src/cli/test/reporters/github/index.ts

```diff
@@ -0,0 +1,51 @@
+import type { Context, Reporter } from '@jest/reporters';
+import type { AggregatedResult } from '@jest/test-result';
+
+import * as GitHub from '../../../../api/github';
+import {
+  buildNameFromEnvironment,
+  enabledFromEnvironment,
+} from '../../../../api/github/environment';
+import { log } from '../../../../utils/logging';
+
+import { generateAnnotationEntries } from './annotations';
+
+export default class GitHubReporter implements Pick<Reporter, 'onRunComplete'> {
+  async onRunComplete(
+    _contexts: Set<Context>,
+    { testResults }: AggregatedResult,
+  ): Promise<void> {
+    if (!enabledFromEnvironment()) {
+      return;
+    }
+
+    try {
+      const entries = generateAnnotationEntries(testResults);
+
+      const build = buildNameFromEnvironment();
+
+      // Create a check run per display name.
+      // Run in series to reduce the likelihood of exceeding GitHub rate limits.
```

Comment:
Just double checking that this was the intent of running in series?

Reply:
Correcto, you wake up early as 👀